components/mdns/Kconfig

```diff
@@ -77,6 +77,20 @@ menu "mDNS"
             Configures period of mDNS timer, which periodically transmits packets
             and schedules mDNS searches.
 
+    config MDNS_IPV4
+        bool "Enable IPv4 for mDNS"
+        default y
+        select LWIP_IPV4
+        help
+            Enable IPv4 for mDNS
+
+    config MDNS_IPV6
+        bool "Enable IPv6 for mDNS"
+        default y
+        select LWIP_IPV6
+        help
+            Enable IPv6 for mDNS
+
```

Comment:
It would be great if we found another solution. If we start doing it like that, then soon most of network protocols start to define their own `XXX_IPV4/6`

Reply:
I added these to the mdns_private.h so that all the sources file can use LWIP_IPV4/LWIP_IPV6.
```
#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 1, 0)
/* CONFIG_LWIP_IPV4 was introduced in IDF v5.1 */
#ifndef LWIP_IPV4
#ifdef CONFIG_LWIP_IPV4
#define LWIP_IPV4 1
#else
#define LWIP_IPV4 0
#endif // CONFIG_LWIP_IPV4
#endif // LWIP_IPV4
#else
/* For IDF v5.0, set LWIP_IPV4 to 1 by default */
#ifndef LWIP_IPV4
#define LWIP_IPV4 1
#endif // LWIP_IPV4
#endif // ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 1, 0)

#ifndef LWIP_IPV6
#ifdef CONFIG_LWIP_IPV6
#define LWIP_IPV6 1
#else
#define LWIP_IPV6 0
#endif // CONFIG_LWIP_IPV6
#endif // LWIP_IPV6
```